### PR TITLE
Thrust voting

### PR DESCRIPTION
**Pull request by griwodz**
Compiling and running on CUDA 8 works now. Tested only on the GTX 980 Ti so far.

CUB code is now compiled for CUDA < 8, Thrust code for CUDA >= 8
Note that CUB code with CUDA 7.5 will still fail on GTX 980 cards and other modern cards.

Test program detection can now use DevIL to load more image formats.

The CI on travis is also updated to test the build with different cuda versions (connected to #49) and upgrading to trusty (connected to #41 )

### CODE REVIEW

File: src/applications/detection/main.cpp [hunk 1]
```diff
@@ -297,6 +301,48 @@ int main(int argc, char** argv)
...
+  if( (ext == ".bmp") ||
+      (ext == ".gif") ||
+      (ext == ".jpg") ||
+      (ext == ".lbm") ||
+      (ext == ".pbm") ||
+      (ext == ".pgm") ||
+      (ext == ".png") ||
+      (ext == ".ppm") ||
+      (ext == ".tga") ||
+      (ext == ".tif") )
```
Comment (fabiencastan) [inline src/applications/detection/main.cpp]: Are these file formats not supported by opencv, which is already in the dependencies?

File: .travis.yml [hunk 2]
```diff
@@ -1,32 +1,44 @@
...
 matrix:
   include:
     - compiler: gcc
-      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc VISUAL_DEBUG=ON CCTAG_SERIALIZE=ON WITH_CUDA=ON
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="7" CUDA_VERSION_MINOR="0" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}-28" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
+    - compiler: gcc
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="7" CUDA_VERSION_MINOR="5" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}-18" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
     - compiler: gcc
-      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc VISUAL_DEBUG=OFF CCTAG_SERIALIZE=ON WITH_CUDA=OFF
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="8" CUDA_VERSION_MINOR="0" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}.61-1" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
```
Comment (fabiencastan) [inline .travis.yml]: Is it not better to add a new variant?
In order to keep one build variant with "VISUAL_DEBUG=ON CCTAG_SERIALIZE=ON"?
To ensure that the code with debug exports is builded once and not broken by new changes.

Comment (simogasp) [inline .travis.yml]: I restored it but not as a variant in the matrix: at the end of the build(s) without cuda, it restarts with those flags. It avoids to explose the matrix of possibilities and also it's faster.

Comment (simogasp) [inline .travis.yml]: incidentally i fixed #41 and #49, so now it also builds clang with cuda, using boost and eigen from the repositories

**Comment by simogasp** [comment]
Before merging maybe update the README/BUILD with the updated info: required/supported CUDA versions, supported HW etc
@griwodz @fabiencastan

**Pull request merged**